### PR TITLE
refactor writeto db to smaller function

### PR DESCRIPTION
**Pull request by yokofly**
I intend to split as 2 prs for #35
the first part is to refactor the writetodb function, I have run some tests, probably the CI will run a more compatible suite.

I conduct a more direct test to verify the temp table: insert a big table, stop with ctrl+c during the insert, and collect the log 
this is the whole log difference(between release 1.2.20 and commit ac8802a30), from the log seems all good
https://www.diffchecker.com/f4YCxD9W/

### CODE REVIEW

**Comment by flarco** [comment]
Thanks, started to test locally but dealing with another issue.
I'm travelling this week so I may not get to it till this weekend or next week.

Also, there are a few instances of the use of `g.Error` that need fixing.

For example, instead of `g.Error("Could not perform upsert from temp: %v", err)` could you use `g.Error(err, "Could not perform upsert from temp")`? It's a wrapper that I use for error logging/stack tracing, if the first param is an`error` type, it treats it differently.

**Comment by yokofly** [comment]
hahaha, let's enjoy the trip first! 
Will adjust the error wrapper.

**Comment by yokofly** [comment]
I locally performed some simple tests with clickhouse to clickhouse

**Comment by flarco** [comment]
Tested, all good 👍 .

File: core/sling/task_run_write.go [hunk 1]
```diff
@@ -133,15 +134,229 @@ func (t *TaskExecution) WriteToFile(cfg *Config, df *iop.Dataflow) (cnt uint64,
...
+		err = g.Error(err, "could not get count from temp table %s", tableTmp.FullName())
+		return 0, err
+	}
+	if cnt != tCnt {
+		err = g.Error(err, "inserted in temp table but table count (%d) != stream count (%d). Records missing/mismatch. Aborting", tCnt, cnt)
+		return 0, err
+	} else if tCnt == 0 && len(sampleData.Rows) > 0 {
+		err = g.Error(err, "Loaded 0 records while sample data has %d records. Exiting.", len(sampleData.Rows))
+		return 0, err
+	}
```
Comment (yokofly) [inline core/sling/task_run_write.go:264]: this code is buggy, sorry about that, I find the `err` will pass to the internal, and got something like 
```
WRN NewError called with nil error:
```
old code
```
	tCnt, _ := tgtConn.GetCount(tableTmp.FullName())
	if cnt != tCnt {
		err = g.Error("inserted in temp table but table count (%d) != stream count (%d). Records missing/mismatch. Aborting", tCnt, cnt)
		return
	} else if tCnt == 0 && len(sampleData.Rows) > 0 {
		err = g.Error("Loaded 0 records while sample data has %d records. Exiting.", len(sampleData.Rows))
		return
	}
```

**Pull request merged**